Shared/Mirror/mirrorReflection.js

```diff
@@ -0,0 +1,35 @@
+//
+//  mirrorReflection.js
+//
+//  Created by Rebecca Stankus on 8/30/17.
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+// This script acts on the reflection box in front of the mirror so that when and avatar
+// enters the area, the mirror will reflect an image via the spectator camera
+
+(function () {
+    print("Starting reflection script...............");
+    var mirrorID, reflectionAreaID;
+    // get id of reflection area and mirror
+    this.preload = function(entityID) {
+        reflectionAreaID = entityID;
+        mirrorID = Entities.getEntityProperties(reflectionAreaID, 'parentID').parentID;
+        print("Reflection area ID is : " + reflectionAreaID);
+        print("Mirror area ID is : " + mirrorID);
+    };
+
+    // when avatar enters reflection area, begin reflecting
+    this.enterEntity = function(entityID){
+        Entities.callEntityMethod(mirrorID, 'toggleMirrorOverlay');
```

Comment:
Can you please make separate messages for mirrorOverlayOn and Off, because I started the script while I was in the zone, and it toggled the mirror on whenever I left the zone. Its more reliable to have 2 separate messages instead of toggle.